smarts/core/smarts.py

```diff
@@ -580,10 +581,16 @@ def switch_control_to_agent(
 
         return vehicle
 
-    def _stop_managing_with_providers(self, vehicle_id: str):
+    def _provider_for_actor(self, actor_id: str) -> Optional[Provider]:
         for provider in self.providers:
-            if provider.manages_actor(vehicle_id):
-                provider.stop_managing(vehicle_id)
+            if provider.manages_actor(actor_id):
+                return provider
```

Comment:
yeah, good point.  Hopefully for now this is ok since there are only a handful of Providers to iterate through, but if we start accumulating them we will probably want some sort of dictionary to track which actors are owned by each provider.

Reply:
Yes, I think the long-term intention would be to let users add their own providers if using SMARTS directly.